toasty/cli.py

```diff
@@ -895,14 +906,232 @@ def view_impl(settings):
         tiler = FitsTiler(coll)
         tiler.tile(cli_progress=True)
 
+    rel_wtml_path = os.path.join(tiler.out_dir, "index_rel.wtml")
+
+    if settings.tile_only:
+        print("WTML:", rel_wtml_path)
+        return
+
     preview_wtml(
-        os.path.join(tiler.out_dir, "index_rel.wtml"),
+        rel_wtml_path,
         browser=settings.browser,
         app_type="research",
         app_url=settings.appurl,
     )
 
 
+def view_tunneled(settings):
+    import random
+    import subprocess
+    from urllib.parse import urlparse, urlunparse
+    from wwt_data_formats.folder import Folder
+    from wwt_data_formats.server import launch_app_for_wtml
+
+    # First: tunnel to the remote host to tile the data and get the path to the
+    # index_rel.wtml.
+    #
+    # If you do `ssh $host $command`, you don't get a login shell, and in some
+    # setups that means that not all environment initialization happens. Which
+    # might mean that the `toasty` command won't be available. We work around
+    # that by avoiding $command and writing an `exec` to stdin. This is gross
+    # and raises issues of escaping funky filenames, but those issues already
+    # exist with SSH command arguments.
+    #
+    # We give SSH `-T` to prevent warnings about not allocating pseudo-TTYs.
+
+    ssh_argv = ["ssh", "-T", settings.tunnel]
+    toasty_argv = ["exec", "toasty", "view", "--tile-only"]
+    if settings.parallelism:
+        toasty_argv += ["--parallelism", str(settings.parallelism)]
+    toasty_argv += settings.paths
```

Comment:
One issue here is that this won't handle file names with spaces, even when putting the name in quotes (which does work running toasty locally). 

I was able to fix this pretty easy on my machine by replacing this line with
`toasty_argv += [f"\"{path}\"" if " " in path else path for path in settings.paths]`

Reply:
Handled with `shlex.quote()`, which is a standard library function for dealing with this stuff.